contrib/snap/snapcraft.yaml

```diff
@@ -0,0 +1,110 @@
+name: fwupd
+version: ##VERSION##
+summary: fwupd
+description: |
+  fwupd.
+
+grade: devel
+confinement: classic
+
+apps:
+  fwupdtool:
+    command: fwupdtool.wrapper
+
+parts:
+  fwupd:
+    plugin: meson
+    meson-parameters: [--prefix=/usr, --libexecdir=/usr/lib, -Dgtkdoc=false, -Dsnap=true]
+    source: .
+    build-packages:
+      - fontconfig
+      - fonts-dejavu
+      - gcab
+      - gettext
+      - gir1.2-pango-1.0
+      - gnutls-bin
+      - gnutls-dev
+      - gobject-introspection
+      - gtk-doc-tools
+      - help2man
+      - libappstream-glib-dev
+      - libarchive-dev
+      - libcairo-dev
+      - libcairo-gobject2
+      - libcolord-dev
+      - libcolorhug-dev
+      - libefivar-dev
+      - libelf-dev
+      - libfreetype6-dev
+      - libfwup-dev
+      - libgcab-dev
+      - libgirepository1.0-dev
+      - libglib2.0-dev
+      - libgpgme11-dev
+      - libgudev-1.0-dev
+      - libgusb-dev
+      - libjson-glib-dev
+      - libpolkit-gobject-1-dev
+      - libsmbios-dev
+      - libsoup2.4-dev
+      - libsqlite3-dev
+      - libtool-bin
+      - locales
+      - meson
+      - pkg-config
+      - policykit-1
+      - python3-gi-cairo
+      - python3-pil
+      - python3-requests
+      - systemd
+      - udev
+      - umockdev
+      - valac
+    stage-packages:
+      - libusb-1.0-0
+      - libappstream-glib8
+      - libarchive13
+      - libc6
+      - libefiboot1
+      - libgusb2
+      - libsmbios-c2
+      - libfwup1
+      - libgpgme11
+      - libelf1
+      - libcolorhug2
+      - libgudev-1.0-0
+      - libjson-glib-1.0-0
+      - liblzma5
+    prime:
```

Comment:
Regarding these package lists, I didn't put them into our package dependency system because I'm not yet sure where this makes most sense in terms of CI.  I wanted to see what you thought.
In the [documentation](https://docs.snapcraft.io/build-snaps/ci-integration) offers stuff related to both Launchpad, CircleCI and Travis CI.

Ideally this should be published to the snap store edge channel and then people can easily install it and we can have latest master always published to edge channel.

Reply:
Not my area of expertise :) ¯\_(ツ)_/¯

Reply:
OK :) I hope that the @lyoncore guys come by and can advise.

Reply:
@superm1 Did you try build.snapcraft.io ? Launchpad is also offering snap automatic building and uploading to store if code is on Launchpad. 

BTW I just setup https://launchpad.net/~timchen119/+snap/fwupd with this branch and found the snapcraft.yaml is not in the snap/ directory, that makes the build failed, is it ok for you to pull it out from contrib/ and make snap/ to top-level? More info on: https://github.com/canonical-websites/build.snapcraft.io/issues/784

Reply:
I don't really want snap a top level directory if I can help it.

Reply:
@timchen119 Actually snapcraft.yaml is dynamic because it needs to contain version information.  Is it possible to call a script before launchpad does that part of the build?  Like how debs could have launchpad packaging recipes?

I didn't use build.snapcraft.io, I just did a local build (calling `contrib/ci/snap.sh`) which prepared build directory and built snap in it.

Reply:
I guess the "alternative" is to have the snapcraft.yaml in a different tree or branch with it placed in the right location, but I'm unclear how you fill in the version information from git in that scenario.

Reply:
@superm1 there is `version-script` field https://github.com/videolan/vlc/blob/master/extras/package/snap/snapcraft.yaml#L2 , another trick to fill version dynamically is to use `snapcraftctl set-version` in the snapcraft.yaml, an example here: https://github.com/ogra1/uboot-tools-snap/blob/master/snap/snapcraft.yaml#L63 ,

Reply:
Thanks, those both look quite workable.

I'll do some experiment with build.snapcraft.io then.

Reply:
Btw it looks that vlc does manage to keep snapcraft.yaml elsewhere in 🌲. How do they get it to build then?